components/Section/Section.tsx

```diff
@@ -0,0 +1,33 @@
+import { ReactNode } from 'react'
+
+interface Props {
+  className?: string
+  p?: string
+  px?: string
+  py?: string
+  pt?: string
+  pb?: string
+  pl?: string
+  pr?: string
+  backgroundColor?: string
+  children: ReactNode
+}
+
+const Section = (props: Props) => {
+  return (
+    <section
+      style={{
```

Comment:
This project uses tailwind and styling should be done using tailwind to maintain the layout throughout the project. Please do the styling with tailwind. You can check other components for examples. You can do this by adding a class according to the condition.

Reply:
https://tailwindcss.com/docs/content-configuration#dynamic-class-names

This is the problem i am trying to find a workaround any suggestion?
So `p-${props.p}` is not working.

Reply:
You can fix this problem by following this guide:

https://tailwindcss.com/docs/content-configuration#class-detection-in-depth

To sum it up briefly, tailwind can't predict what's to come. However, if you send it as a key in an object, there is no problem because it scans that object.

There is a similar usage in Title component, I definitely recommend you to have a look

Reply:
As far as I understand in the Title component it works because there are already established tailwind classes in `.css` which would be included in the output `css file` of tailwind.

Problem with the approach I tried is html gets the class that tsx created but there is no such class in output of tailwind. It will work if I create types like this and use tailwind class correspond to them:
```ts
type Size =  "sm"| "md" | "lg" | "xl"
```
But padding values won't be dynamic that way. Please correct me if I get something wrong.

Reply:
yes it is not dynamic. In order to produce a consistent design throughout the project, it is not really important that it is dynamic. It would be an inconsistent design pattern for one section to have a padding of 1rem and another to have 1.1rem. It's okay if it's static.